packages/@aws-cdk/aws-codecommit/lib/pipeline-action.ts

```diff
@@ -54,11 +54,15 @@ export class PipelineSourceAction extends codepipeline.SourceAction {
       configuration: {
         RepositoryName: props.repository.repositoryName,
         BranchName: props.branch || 'master',
-        PollForSourceChanges: props.pollForSourceChanges !== undefined ? props.pollForSourceChanges : true
+        PollForSourceChanges: props.pollForSourceChanges || false,
       },
       outputArtifactName: props.outputArtifactName
     });
 
+    if (props.pollForSourceChanges !== true) {
+      props.repository.onCommit('CodePipeline', props.stage.pipelineAsEventTarget, props.branch || 'master');
```

Comment:
Again, codify the default for `branch` once in your code:

```ts
const branch = props.branch || 'master';
```

Reply:
Again, can't use a variable here (`super` call, yada yada yada). The only alternative is a function, and that seemed like overkill to me here, hence the duplication.

Reply:
Yada yada you can ;-)